script/dryrun

```diff
@@ -35,33 +37,60 @@ skip={
     "sql/telemetry_derived/addons_v3.sql",
     "sql/telemetry_derived/addons_aggregates_v3.sql",
 }
-for sql in glob.glob('sql/*/*.sql'):
-    if sql in skip:
-        continue
-    pad = ' ' * (60 - len(sql))
+
+
+def worker_entrypoint(sqlfile):
+    sql = open(sqlfile).read()
     try:
-        print(sql, end=pad)
         r = urlopen(
             Request(
                 DRY_RUN_URL,
-                headers = {
-                    'Content-Type': 'application/json'
-                },
-                data=json.dumps({
-                    'dataset': os.path.basename(os.path.dirname(sql)),
-                    'query': open(sql).read(),
-                }).encode('utf8'),
-                method='POST'
-            ))
+                headers={"Content-Type": "application/json"},
+                data=json.dumps(
+                    {
+                        "dataset": os.path.basename(os.path.dirname(sqlfile)),
+                        "query": sql,
+                    }
+                ).encode("utf8"),
+                method="POST",
+            )
+        )
     except Exception as e:
-        print('ERROR\n', e)
-        exitcode = 1
+        print(f"{sqlfile:59} ERROR\n", e)
+        return False
+    response = json.load(r)
+    if "errors" in response and len(response["errors"]) == 1:
+        error = response["errors"][0]
+    else:
+        error = None
+    if response["valid"]:
+        print(f"{sqlfile:59} OK")
+    elif (
+        error
+        and error["code"] == 403
+        and "does not have bigquery.tables.create permission for dataset"
+        in error["message"]
+    ):
+        # We want the dryrun service to only have read permissions, so
+        # we expect CREATE VIEW and CREATE TABLE to throw specific
+        # exceptions.
+        print(f"{sqlfile:59} OK, but with insufficient access to create table/view")
```

Comment:
Added this special case for dryrun so that the service doesn't need permission to create views. I ended up refactoring the script to allow parallel checks, though, since it was taking quite a while to run with all the new definitions.